homeassistant/components/risco/__init__.py

```diff
@@ -50,9 +51,25 @@ class LocalData:
     """A data class for local data passed to the platforms."""
 
     system: RiscoLocal
-    zone_updates: dict[int, Callable[[], Any]] = field(default_factory=dict)
+    zone_entities: dict[int, list[Entity]] = field(default_factory=dict)
     partition_updates: dict[int, Callable[[], Any]] = field(default_factory=dict)
 
+    def add_zone_entity(self, zone_id: int, entity: Entity) -> Callable[[], None]:
+        """Add an entity to be updated when the matching zone is updated."""
+        self.zone_entities.setdefault(zone_id, [])
+        self.zone_entities[zone_id].append(entity)
```

Comment:
Don't store the entity in a shared container. Use our dispatch helper to signal the entity from outside if needed.

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/dispatcher.py

It's ok to store the id of the entity, eg the zone_id.